server/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -820,6 +833,11 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
         MetaDataDiff(StreamInput in) throws IOException {
             clusterUUID = in.readString();
             version = in.readLong();
+            if (in.getVersion().onOrAfter(Version.V_7_0_0)) {
+                coordinationMetaData = new CoordinationMetaData(in);
+            } else {
+                coordinationMetaData = CoordinationMetaData.EMPTY_META_DATA;
```

Comment:
Depending on how we plan to do rolling upgrades, this may or may not be safe, and might need backporting. We're not sure yet. Could you leave a `// TODO check that this is safe` here so we know to revisit it later on.

Reply:
https://github.com/elastic/elasticsearch/pull/35625/commits/9006e32fcb0f385af8d8dd91fda67c810b0a61c6